src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java

```diff
@@ -34,34 +44,131 @@
+        populatePostponementSscsCaseData();
+
+        sscsCaseData.setPostponementRequest(PostponementRequest.builder().actionPostponementRequestSelected("grant")
+                .listingOption("readyToList").build());
+
+        PreSubmitCallbackResponse<SscsCaseData> response =
+                handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        verify(footerService).createFooterAndAddDocToCase(eq(expectedDocument.getValue().getDocumentLink()), any(),
+                eq(POSTPONEMENT_REQUEST_DIRECTION_NOTICE), any(), any(), eq(null), eq(null));
+        assertThat(response.getData().getInterlocReviewState(), is(nullValue()));
+        assertThat(response.getData().getInterlocReferralReason(), is(nullValue()));
+        assertThat(response.getData().getState(), is(State.READY_TO_LIST));
+        assertThat(response.getData().getAppeal().getHearingOptions().getExcludeDates().stream()
+                .anyMatch(excludeDate -> excludeDate.getValue().getStart()
+                        .equals(LocalDate.now().plusDays(1).toString())), is(true));
+        assertThat(response.getData().getSscsDocument(), is(not(empty())));
+        assertThat(response.getData().getPostponementRequest().getUnprocessedPostponementRequest(), is(YesNo.NO));
+        assertThat(response.getData().getDwpState(), is(DwpState.DIRECTION_ACTION_REQUIRED.getId()));
+    }
+
+    @Test
+    public void givenAGrantedPostponementAndNotListable_thenClearReviewStateAndReferralReasonAndFlagAndAddNoteAndUnavailabilityUpdatedAndDwpStateAndDecisionDocAdded() {
+        populatePostponementSscsCaseData();
+
+        sscsCaseData.setPostponementRequest(PostponementRequest.builder().actionPostponementRequestSelected("grant")
+                .listingOption("notListable").build());
+
+        PreSubmitCallbackResponse<SscsCaseData> response =
+                handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        verify(footerService).createFooterAndAddDocToCase(eq(expectedDocument.getValue().getDocumentLink()), any(),
+                eq(POSTPONEMENT_REQUEST_DIRECTION_NOTICE), any(), any(), eq(null), eq(null));
+        assertThat(response.getData().getInterlocReviewState(), is(nullValue()));
+        assertThat(response.getData().getInterlocReferralReason(), is(nullValue()));
+        assertThat(response.getData().getState(), is(State.NOT_LISTABLE));
+        assertThat(response.getData().getAppeal().getHearingOptions().getExcludeDates().stream()
+                .anyMatch(excludeDate -> excludeDate.getValue().getStart()
+                        .equals(LocalDate.now().plusDays(1).toString())), is(true));
+        assertThat(response.getData().getPostponementRequest().getUnprocessedPostponementRequest(), is(YesNo.NO));
+        assertThat(response.getData().getDwpState(), is(DwpState.DIRECTION_ACTION_REQUIRED.getId()));
+    }
+
+    private void populatePostponementSscsCaseData() {
```

Comment:
Add a test with multiple hearings in future

Reply:
It can never happen